msal/oauth2cli/oauth2.py

```diff
@@ -35,13 +34,12 @@ def __init__(
             self,
             server_configuration,  # type: dict
             client_id,  # type: str
+            http_client,  # type: HttpClient
```

Comment:
According to the current implementation, it is an http_client which can also be a session, session will hold true for session objects only which is what we historically do. But http_client is what we expect, if the passed http client is a session object it still a http_client as we assume it is supposed to have qualities similar to that http_client interface.

Reply:
This entire file looks in good shape. And, as part of the final cross-check, I compare it to my async branch, and notice some line(s) are missing/different. A few of them are actually important.
At this point, rather than pointing them out one-by-one in this PR (which would be very time-consuming) and then you need to hand-pick them, we can probably just download [this file](https://raw.githubusercontent.com/AzureAD/microsoft-authentication-library-for-python/oauth2-session/msal/oauth2cli/oauth2.py) into your laptop, and then, either run a "diff" to see the difference, or you can just use it and see the tests presumably still pass.
(Note that I've changed my previous `session` naming to adopt your `http_client`, so you can use it as a drop-in replacement.)

I also adopt your `http_client` naming to replace the `session` in my another working branch.